### DISCUSSION TITLE
Multiplayer doesn't start if an overlay is opened in the loading screen

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#23602] [Q&A] [answered]
# Steps to reproduce
1. Create a multiplayer room
2. Start the match
3. In the loading screen, open an overlay
## Expected result
As multiplayer is real-time, the overlay should close and gameplay should start.
## Actual result
The overlay seems to close and gameplay starts but only after waiting for 30 seconds. Is this intended?

**Comment by bdach** [comment, marked as answer]
This is semi-intentional. The reason for this is that doing anything on the loading screen prevents the load process from progressing to gameplay, up to 30 seconds, at which point the [gameplay start is forced server-side](https://github.com/ppy/osu-server-spectator/blob/0debcef2b35fac46c6e749b8c80787093d6d1277/osu.Server.Spectator/Hubs/MultiplayerHubContext.cs#L24C22-L27) and any players that haven't loaded yet are booted from the game.

I think we still want this to be the case, to let players adjust their settings on the loader screen?

**Reply by peppy** [reply]
Yeah, it's definitely intended. Arguable we could force close the overlays when the game starts as it wouldn't cause any state loss?

**Reply by soopax** [reply]
Nevermind, I think I misunderstood how it works. Sorry.
